### DISCUSSION TITLE
Couldn't run Dupe Marker Detector

### DISCUSSION THREAD

**Discussion by pthanhtu92** [stashapp/stash#5303] [Q&A] [answered]
I tried to run Mark task with Dupe Marker Detector plugin but I keep getting this error. I did try to do this command. `python -m pip install --upgrade certifi`. I didn't work. Could anyone help me with this please? This is the error.
`INFO[2024-09-26 13:15:17] using config file: config.yml
INFO[2024-09-26 13:15:28] [InitHWSupport] Supported HW codecs [2]:
        H264 NVENC HQ profile - h264_nvenc
        H264 NVENC - h264_nvenc
INFO[2024-09-26 13:15:28] stash version: v0.27.0 (c45ae068) - Official Build - 2024-09-23 03:00:24
INFO[2024-09-26 13:15:28] stash is listening on 0.0.0.0:9999
INFO[2024-09-26 13:15:28] stash is running at https://localhost:9999/
INFO[2024-09-26 13:15:29] Version v0.27.0 (c45ae068) is already the latest released
2024-09-26T13:16:33.6293729-04:00 INF Response: 200 OK httpRequest={"proto":"HTTP/1.1","remoteIP":"127.0.0.1:42771","requestID":"Bee/XZDuqvVRu2-000006","requestMethod":"POST","requestPath":"/graphql","requestURL":"https://127.0.0.1:9999/graphql"} httpResponse={"bytes":54,"elapsed":32.9492,"status":200} service=stash
2024/09/26 13:16:34 http: TLS handshake error from 127.0.0.1:42775: local error: tls: bad record MAC
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Could not connect to Stash at https://127.0.0.1:9999/graphql
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)')))
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last):
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 467, in _make_request
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self._validate_conn(conn)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 1099, in _validate_conn
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     conn.connect()
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 653, in connect
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     sock_and_verified = _ssl_wrap_socket_and_match_hostname(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 806, in _ssl_wrap_socket_and_match_hostname
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ssl_sock = ssl_wrap_socket(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\ssl_.py", line 465, in ssl_wrap_socket
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\ssl_.py", line 509, in _ssl_wrap_socket_impl
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 455, in wrap_socket
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self.sslsocket_class._create(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 1042, in _create
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self.do_handshake()
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 1320, in do_handshake
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self._sslobj.do_handshake()
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] During handling of the above exception, another exception occurred:
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last):
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 793, in urlopen
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     response = self._make_request(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 491, in _make_request
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise new_e
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] urllib3.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] The above exception was the direct cause of the following exception:
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last):
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 486, in send
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     resp = conn.urlopen(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 847, in urlopen
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     retries = retries.increment(
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]               ^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\retry.py", line 515, in increment
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)')))
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] During handling of the above exception, another exception occurred:
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last):
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "Z:\stash\plugins\community\dupeMarker/dupeMarker.py", line 8, in <module>
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     stash = StashInterface(FRAGMENT["server_connection"])
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 52, in __init__
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self.version = self.stash_version()
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                    ^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 224, in stash_version
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     result = self.call_GQL("query StashVersion{ version { build_time hash version } }")
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 221, in call_GQL
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self._GQL(query, variables)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\classes.py", line 224, in _GQL
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     response = self.s.post(self.url, json=json_request)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 637, in post
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self.request("POST", url, data=data, json=json, **kwargs)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 589, in request
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     resp = self.send(prep, **send_kwargs)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 703, in send
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     r = adapter.send(request, **kwargs)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 517, in send
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise SSLError(e, request=request)
ERRO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)')))
ERRO[2024-09-26 13:16:34] Plugin returned error: exit status 1
`

**Comment by stg-annon** [comment, marked as answer]
```
INFO[2024-09-26 13:15:17] using config file: config.yml 
INFO[2024-09-26 13:15:28] [InitHWSupport] Supported HW codecs [2]: H264 NVENC HQ profile - h264_nvenc H264 NVENC - h264_nvenc 
INFO[2024-09-26 13:15:28] stash version: v0.27.0 (c45ae068) - Official Build - 2024-09-23 03:00:24 
INFO[2024-09-26 13:15:28] stash is listening on 0.0.0.0:9999 
INFO[2024-09-26 13:15:28] stash is running at https://localhost:9999/ 
INFO[2024-09-26 13:15:29] Version v0.27.0 (c45ae068) is already the latest released 2024-09-26T13:16:33.6293729-04:00 INF Response: 200 OK httpRequest={"proto":"HTTP/1.1","remoteIP":"127.0.0.1:42771","requestID":"Bee/XZDuqvVRu2-000006","requestMethod":"POST","requestPath":"/graphql","requestURL":"https://127.0.0.1:9999/graphql"} httpResponse={"bytes":54,"elapsed":32.9492,"status":200} service=stash 2024/09/26 13:16:34 http: TLS handshake error from 127.0.0.1:42775: local error: tls: bad record MAC 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Could not connect to Stash at https://127.0.0.1:9999/graphql 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)'))) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last): 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 467, in _make_request 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self._validate_conn(conn) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 1099, in _validate_conn 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     conn.connect() 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 653, in connect 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     sock_and_verified = _ssl_wrap_socket_and_match_hostname( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 806, in _ssl_wrap_socket_and_match_hostname 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ssl_sock = ssl_wrap_socket( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\ssl_.py", line 465, in ssl_wrap_socket 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\ssl_.py", line 509, in _ssl_wrap_socket_impl 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return ssl_context.wrap_socket(sock, server_hostname=server_hostname) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 455, in wrap_socket 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self.sslsocket_class._create( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 1042, in _create 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self.do_handshake() 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\ssl.py", line 1320, in do_handshake 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self._sslobj.do_handshake() 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] During handling of the above exception, another exception occurred: 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last): 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 793, in urlopen 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     response = self._make_request( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 491, in _make_request 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise new_e 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] urllib3.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] The above exception was the direct cause of the following exception: 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last): 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 486, in send 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     resp = conn.urlopen( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 847, in urlopen 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     retries = retries.increment( 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]               ^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\retry.py", line 515, in increment 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type] 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)'))) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] During handling of the above exception, another exception occurred: 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] Traceback (most recent call last): 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "Z:\stash\plugins\community\dupeMarker/dupeMarker.py", line 8, in <module> 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     stash = StashInterface("server_connection"]) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 52, in __init__ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     self.version = self.stash_version() 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                    ^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 224, in stash_version 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     result = self.call_GQL("query StashVersion{ version { build_time hash version } }") 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\stashapp.py", line 221, in call_GQL 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self._GQL(query, variables) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\stashapi\classes.py", line 224, in _GQL 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     response = self.s.post(self.url, json=json_request) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 637, in post 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     return self.request("POST", url, data=data, json=json, **kwargs) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 589, in request 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     resp = self.send(prep, **send_kwargs) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 703, in send 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     r = adapter.send(request, **kwargs) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]   File "C:\Users\ngoho\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 517, in send 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector]     raise SSLError(e, request=request) 
INFO[2024-09-26 13:16:34] [Plugin / Dupe Marker Detector] requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=9999): Max retries exceeded with url: /graphql (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)'))) 
INFO[2024-09-26 13:16:34] Plugin returned error: exit status 1
```

based on the provided error you are using a self signed SSL certificate and the library that the plugin uses requires the certificate to be signed or a flag provided to ignore self signed certs. for example [this line](https://github.com/stashapp/CommunityScripts/blob/bea9ce1a2e0af97be5481289c2a79d7ebf39c35e/plugins/dupeMarker/dupeMarker.py#L8) would need to be 

```py
stash = StashInterface(FRAGMENT["server_connection"], verify_ssl=False)
``` 



 this would be more of a question for the community repo for the plugin https://github.com/stashapp/CommunityScripts

**Reply by pthanhtu92** [reply]
Thank you!!
